packages/openmrs/test/index.js

```diff
@@ -47,3 +59,239 @@ describe('execute', () => {
     });
   });
 });
+
+describe('request', () => {
+  it('should search for a patient', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah 1' },
+        method: 'GET',
+      })
+      .reply(200, { results: [{ display: 'Sarah 1' }] }, { ...jsonHeaders });
+    const state = { configuration };
+
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah 1' }
+    );
+
+    expect(body.results[0].display).to.eql('Sarah 1');
+  });
+  it('should auto-fetch patients with a limit', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 1' }],
+          links: [
+            {
+              rel: 'next',
+              uri: 'https://fn.openmrs.org/ws/rest/v1/patient?q=Sarah&limit=1&startIndex=1',
+              resourceAlias: null,
+            },
+          ],
+        },
+        { ...jsonHeaders }
+      );
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1, startIndex: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 2' }],
+        },
+        { ...jsonHeaders }
+      );
+
+    const state = { configuration };
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah', limit: 1 }
+    );
+    expect(body.results[0].display).to.eql('Sarah 1');
+    expect(body.results[1].display).to.eql('Sarah 2');
+  });
+
+  it('should not auto-fetch if the user sets startIdex', async () => {
+    testServer
+      .intercept({
+        path: '/ws/rest/v1/patient',
+        query: { q: 'Sarah', limit: 1, startIndex: 1 },
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          results: [{ display: 'Sarah 2' }],
+        },
+        { ...jsonHeaders }
+      );
+
+    const state = { configuration };
+    const { body } = await request(
+      state,
+      'GET',
+      '/ws/rest/v1/patient',
+      {},
+      { q: 'Sarah', limit: 1, startIndex: 1 }
```

Comment:
If you remove `startIndex` from this test,, does it still pass?

I'll bet it does - but it shouldn't!

You need to set up the interceptor to return a next page. That way you are properly testing that your adaptor code will ignore the next page IF the startIndex is set

Reply:
I have added the interceptor for the next page

Reply:
Ah, ok, but now we still need to assert that only one page was actually fetched. You need to check that `body.results.length === 1`

Reply:
Done, i have added the assertion for results to be equal  to 1